crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1634,11 +1635,11 @@ impl AuthorityStore {
                 (init.0 + result.0, init.1 + result.1)
             })
         });
-        let mut layout_resolver = TypeLayoutResolver::new(move_vm, self.as_ref());
+        let mut layout_resolver = executor.type_layout_resolver(Box::new(self.as_ref()));
```

Comment:
Noticed something here that I didn't fully understand.  Here, we are using the `AuthorityStore` directly as the store that `TypeLayoutResolver` queries for packages when resolving type layouts.  Above, we use the `package_cache`.  Why do we not use the package cache here? cc @lxfind .

Reply:
That would be a question for @dariorussi. I think he added the layout resolver here.

Reply:
not sure what to say here. I changed the code as it was, in other words it was using `self` before instead of the package cache and I left it that way. I wanted to raise the issue in the PR but then that PR got crazy and totally forgot.
Looking at the code it feels we should be able to use the cache which is really a wrapper around `self`.
Sorry I have no more to add here

Reply:
https://github.com/MystenLabs/sui/pull/11529/files#diff-b07ea487376f67264505e88e54693c63f78e1720d0f9c928119d032a45925223 introduced this line, so it wasn't there before.

Reply:
yes I know, however I was pointing out that the code changed as follows
```
-            total_sui += object.get_total_sui(self).unwrap() - object.storage_rebate;
+            total_sui +=
+                object.get_total_sui(&mut layout_resolver).unwrap() - object.storage_rebate;
+        }
```
in other words the call to `get_total_sui` was there with `self` and not `package_cache` and that is how I left it (maybe by mistake and should have "fixed" it).
So the question asked stands and it's whether that should take the package cache or not.
I think it should and wonder what is your take

Reply:
Based on the resolution here (TBD), I'll also deal with this as a separate follow-up PR.  It does seem like we could use the package cache here though.